Windows/macOS test results are no longer available, but I got this from the Linux results:
```
FixedPointConversion_Debug64_ToInt-2849e2.o:function main:
error: undefined reference to '__extendhfsf2'
```

--- a/stdlib/public/runtime/Float16Support.cpp
+++ b/stdlib/public/runtime/Float16Support.cpp
@@ -31,7 +31,7 @@
 // Android NDK <r21 do not provide `__aeabi_d2h` in the compiler runtime,
 // provide shims in that case.
 #if (defined(__ANDROID__) && defined(__ARM_ARCH_7A__) && defined(__ARM_EABI__)) || \
-  ((defined(__i386__) || defined(__i686__) || defined(__x86_64__)) && !defined(__APPLE__) && !defined(__linux__))
+  ((defined(__i386__) || defined(__i686__) || defined(__x86_64__)) && !defined(__APPLE__))
 
 #include "swift/shims/Visibility.h"
 
@@ -46,6 +46,20 @@ static float fromEncoding(unsigned int e) {
   float f;
   static_assert(sizeof f == sizeof e, "float and int must have the same size");
   __builtin_memcpy(&f, &e, sizeof f);
+  return f;
+}
+
+static unsigned short toEncoding(_Float16 f) {
+  unsigned short s;
+  static_assert(sizeof s == sizeof f, "_Float16 and short must have the same size");
+  __builtin_memcpy(&s, &f, sizeof f);
+  return s;
+}
+
+static _Float16 fromEncoding(unsigned short s) {
+  _Float16 f;
+  static_assert(sizeof s == sizeof f, "_Float16 and short must have the same size");
+  __builtin_memcpy(&f, &s, sizeof f);
   return f;
 }
 
@@ -126,7 +140,7 @@ SWIFT_RUNTIME_EXPORT unsigned short __gnu_f2h_ieee(float f) {
 //
 // Note that F16C doesn't provide this operation, so we still need a software
 // implementation on those cores.
-SWIFT_RUNTIME_EXPORT unsigned short __truncdfhf2(double d) {
+SWIFT_RUNTIME_EXPORT _Float16 __truncdfhf2(double d) {
   // You can't just do (half)(float)x, because that makes the result
   // susceptible to double-rounding. Instead we need to make the first
   // rounding use round-to-odd, but that doesn't exist on x86, so we have
@@ -148,7 +162,7 @@ SWIFT_RUNTIME_EXPORT unsigned short __truncdfhf2(double d) {
     if (fabs < dabs) e |= 1;
     f = fromEncoding(e);
   }
-  return __gnu_f2h_ieee(f);
+  return fromEncoding(__gnu_f2h_ieee(f));
 }
 
 // Convert from Float16 to long double.
@@ -160,18 +174,18 @@ SWIFT_RUNTIME_EXPORT unsigned short __truncdfhf2(double d) {
 //
 // There's no risk of rounding problems from the double conversion, because
 // we're extending.
-SWIFT_RUNTIME_EXPORT long double __extendhfxf2(short h) {
-  return __gnu_h2f_ieee(h);
+SWIFT_RUNTIME_EXPORT long double __extendhfxf2(_Float16 h) {
+  return __gnu_h2f_ieee(toEncoding(h));
 }
 
 // This is just an alternative name for __gnu_h2f_ieee
-SWIFT_RUNTIME_EXPORT float __extendhfsf2(unsigned short h) {
-  return __gnu_h2f_ieee(h);
+SWIFT_RUNTIME_EXPORT float __extendhfsf2(_Float16 h) {
+  return __gnu_h2f_ieee(toEncoding(h));
 }
 
 // Same again but for __gnu_f2h_ieee
-SWIFT_RUNTIME_EXPORT unsigned short __truncsfhf2(float f) {
-  return __gnu_f2h_ieee(f);
+SWIFT_RUNTIME_EXPORT _Float16 __truncsfhf2(float f) {
+  return fromEncoding(__gnu_f2h_ieee(f));
 }
 
 #if defined(__ARM_EABI__)

--- a/stdlib/public/runtime/Float16Support.cpp
+++ b/stdlib/public/runtime/Float16Support.cpp
@@ -15,6 +15,7 @@
 // __gnu_h2f_ieee
 // __gnu_f2h_ieee
 // __truncdfhf2
+// __extendhfxf2
 //
 // On Darwin platforms, these are provided by the host compiler-rt, but we
 // can't depend on that everywhere, so we have to provide them in the Swift
@@ -147,6 +148,29 @@ SWIFT_RUNTIME_EXPORT unsigned short __truncdfhf2(double d) {
     if (fabs < dabs) e |= 1;
     f = fromEncoding(e);
   }
+  return __gnu_f2h_ieee(f);
+}
+
+// Convert from Float16 to long double.
+//
+// Since Float32 covers the entire range
+// of Float16 values and since we already know how to convert Float32 to long
+// double (which, at least on x86, doesn't involve function calls), we just
+// let the compiler do the latter part for us.
+//
+// There's no risk of rounding problems from the double conversion, because
+// we're extending.
+SWIFT_RUNTIME_EXPORT long double __extendhfxf2(short h) {
+  return __gnu_h2f_ieee(h);
+}
+
+// This is just an alternative name for __gnu_h2f_ieee
+SWIFT_RUNTIME_EXPORT float __extendhfsf2(unsigned short h) {
+  return __gnu_h2f_ieee(h);
+}
+
+// Same again but for __gnu_f2h_ieee
+SWIFT_RUNTIME_EXPORT unsigned short __truncsfhf2(float f) {
   return __gnu_f2h_ieee(f);
 }
 

